pkg/config/store/contextch.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+	"sync"
+)
+
+// contextCh is a utility type of outgoing channel with context. It can send messages
+// as long as the context is valid.
+type contextCh struct {
+	ctx context.Context
+	ch  chan<- BackendEvent
+}
+
+// ContextChList manages the list of ContextCh.
+type ContextChList struct {
+	mu   sync.RWMutex
+	list []*contextCh
+}
+
+// NewContextChList creates a new list of ContextCh.
+func NewContextChList() *ContextChList {
```

Comment:
No need for a new method, delete and replace all occurrences with `&ContextChList{}`.

Reply:
ah, right... removed.